### PR TITLE
Extract OrderingScheme#toLocalProperties

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java [hunk 1]
```diff
@@ -565,12 +551,8 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             Map<Symbol, NullableValue> constants = entries.stream()
                     .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-            ImmutableList.Builder<SortingProperty<Symbol>> localProperties = ImmutableList.builder();
-            if (node.getOrderingScheme().isPresent()) {
-                node.getOrderingScheme().get().getOrderBy().stream()
-                        .map(column -> new SortingProperty<>(column, node.getOrderingScheme().get().getOrdering(column)))
-                        .forEach(localProperties::add);
-            }
+            ImmutableList.Builder<LocalProperty<Symbol>> localProperties = ImmutableList.builder();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java:554]: rewrite:
```
List<LocalProperty<Symbol>> localProperties = node.getOrderingScheme()
  .map(OrderingScheme::toLocalProperties);
```

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java [hunk 2]
```diff
@@ -334,9 +330,7 @@ public ActualProperties visitTopNRanking(TopNRankingNode node, List<ActualProper
 
             ImmutableList.Builder<LocalProperty<Symbol>> localProperties = ImmutableList.builder();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java:331]: rewrite
```
List<LocalProperty<Symbol>> localProperties = node.getOrderingScheme()
  .map(OrderingScheme::toLocalProperties);
```

Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java:331]: there's a `localProperties.add(new GroupingProperty<>(node.getPartitionBy()));` too

File: core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java:554]: In https://github.com/trinodb/trino/pull/7589, I use the builder to add local input properties too

**Review by sopel39 — APPROVED** [review]

**Pull request merged**